frontend/src/component/suggestChanges/SuggestedChangeOverview/SuggestedChangeTimeline/SuggestedChangeTimeline.tsx

```diff
@@ -1,52 +1,136 @@
 import { FC } from 'react';
+import { styled } from '@mui/material';
 import { Box, Paper } from '@mui/material';
 import Timeline from '@mui/lab/Timeline';
 import TimelineItem, { timelineItemClasses } from '@mui/lab/TimelineItem';
 import TimelineSeparator from '@mui/lab/TimelineSeparator';
 import TimelineDot from '@mui/lab/TimelineDot';
 import TimelineConnector from '@mui/lab/TimelineConnector';
 import TimelineContent from '@mui/lab/TimelineContent';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { SuggestChangesetState } from '../suggestChanges.types';
+interface ISuggestChangeTimelineProps {
+    state: SuggestChangesetState;
+}
 
-export const SuggestedChangeTimeline: FC = () => {
+interface ITimelineData {
+    title: string;
+    active: boolean;
+}
+
+const StyledPaper = styled(Paper)(({ theme }) => ({
+    marginTop: theme.spacing(2),
+    borderRadius: `${theme.shape.borderRadiusLarge}px`,
+}));
+
+const StyledBox = styled(Box)(({ theme }) => ({
+    padding: theme.spacing(2),
+    marginBottom: `-${theme.spacing(4)}`,
+}));
+
+const StyledTimeline = styled(Timeline)(() => ({
+    [`& .${timelineItemClasses.root}:before`]: {
+        flex: 0,
+        padding: 0,
+    },
+}));
+
+export const SuggestedChangeTimeline: FC<ISuggestChangeTimelineProps> = ({
+    state,
+}) => {
+    const createTimeLineData = (
+        state: SuggestChangesetState
+    ): ITimelineData[] => {
+        const steps: SuggestChangesetState[] = [
+            'Draft',
+            'In review',
+            'Approved',
+            'Applied',
+        ];
+
+        return steps.map(step => ({
+            title: step,
+            active: step === state,
+        }));
+    };
+
+    const renderTimeline = () => {
+        const data = createTimeLineData(state);
+        const activeIndex = data.findIndex(item => item.active);
+        const hasActiveStep = activeIndex !== -1;
+
+        if (state === 'Cancelled') {
+            return createCancelledTimeline(data);
+        }
+
+        return createTimeline(data, hasActiveStep, activeIndex);
+    };
+
+    return (
+        <StyledPaper elevation={0}>
+            <StyledBox>
+                <StyledTimeline>{renderTimeline()}</StyledTimeline>
+            </StyledBox>
+        </StyledPaper>
+    );
+};
+
+const createTimeline = (
+    data: ITimelineData[],
+    hasActiveStep: boolean,
+    activeIndex: number
```

Comment:
Do these two parameters potentially conflict? What happens if `hasActiveStep` is `true`, but the `activeIndex` is `-1`? Or what if `hasActiveStep` is false, but `activeIndex` is 0? Could that be covered by a union type? Something like:
`type Index = "inactive" | number` (or even `type Index = number?`) or the like?

Reply:
Possibly, will check it out.